js/lib/baseDomain.js

```diff
@@ -9,14 +9,27 @@ const LRUCache = require('lru_cache/core').LRUCache
 
 let cachedBaseDomain = new LRUCache(50)
 
+const checkASCII = function (str) {
+  if (typeof str !== 'string') {
+    return false
+  }
+  for (let i = 0; i < str.length; i++) {
+    if (str.charCodeAt(i) > 127) {
```

Comment:
I think charCodeAt is implemented to UCS-2 even know the strings are UTF-16.  So for example a single unicode symbol could be for charCodeAt(0) and charCodeAt(1).  Are there any code units that have their second byte < 127?  Maybe there is a utility function we can use instead for this since these types of checks can easily go wrong? On a side note I think our TP and adblock libs probably need fixes like this.